pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1055,8 +1061,12 @@ async def on_package(pkg):
                 Args:
                     pkg (createrepo_c.Package): A completed createrepo_c package.
                 """
-                if self.mirror and (pkg.location_base or ".." in pkg.location_href):
-                    raise ValueError(MIRROR_INCOMPATIBLE_REPO_ERR_MSG)
+                if self.mirror:
+                    uses_base_url = pkg.location_base
+                    illegal_relative_path = pkg.location_href.count("../") > self.namespace_depth
```

Comment:
And here. Or - maybe encapsulate into a _is_illegal_path(pkg.location_href) method on the object and use it both places?